nodes/surface/bevel_curve.py

```diff
@@ -0,0 +1,268 @@
+        ('L', "Curve length", "Scaling along curve is proportional to curve segment length", 1)
+    ]
+
+    length_mode : EnumProperty(
+        name = "Samples distribution along path",
+        items = length_modes,
+        default = 'T',
+        update = update_sockets)
+
+    def draw_buttons(self, context, layout):
+        layout.prop(self, 'curve_mode', expand=True)
+        if self.curve_mode == 'NURBS':
+            layout.prop(self, 'precision_method', text='Precision')
+        layout.prop(self, "algorithm")
+        layout.label(text="Orientation:")
+        row = layout.row()
+        row.prop(self, "orient_axis", expand=True)
+        row.enabled = self.algorithm not in {SvBendAlongCurveField.ZERO, SvBendAlongCurveField.FRENET, SvBendAlongCurveField.TRACK_NORMAL}
+
+        if self.algorithm == 'track':
+            layout.prop(self, "up_axis")
+        layout.label(text="Samples distribution:")
+        layout.prop(self, 'length_mode', text='')
+
+    def sv_init(self, context):
+        self.inputs.new('SvCurveSocket', "Path")
+        self.inputs.new('SvCurveSocket', "Profile")
+        self.inputs.new('SvCurveSocket', "Taper")
+        self.inputs.new('SvStringsSocket', "Resolution").prop_name = 'resolution'
+        self.inputs.new('SvStringsSocket', "ProfileCopies").prop_name = 'taper_samples'
+        self.inputs.new('SvStringsSocket', "TaperRefine").prop_name = 'taper_refine'
+        self.inputs.new('SvStringsSocket', "TaperCopies").prop_name = 'profile_samples'
+        self.outputs.new('SvSurfaceSocket', "Surface")
+        self.update_sockets(context)
+
+    def _make_unit_taper(self, path, profile):
+        orient_axis = self._get_orient_axis_idx()
+        x_axis = (orient_axis + 1) % 3
+
+        if profile is not None:
+            profile_u_min = profile.get_u_bounds()[0]
+            profile_start = profile.evaluate(profile_u_min)
+            profile_start[orient_axis] = 0.0
+            radius = np.linalg.norm(profile_start)
+            profile_degree = profile.get_degree()
+        else:
+            radius = 1.0
+            profile_degree = 2
+
+        path_u_min, path_u_max = path.get_u_bounds()
+        path_start = path.evaluate(path_u_min)
+        path_end = path.evaluate(path_u_max)
+
+        z_min = path_start[orient_axis]
+        z_max = path_end[orient_axis]
+
+        p1 = np.zeros((3,), dtype=np.float64)
+        p1[x_axis] = radius
+        p1[orient_axis] = z_min
+
+        p2 = np.zeros((3,), dtype=np.float64)
+        p2[x_axis] = radius
+        p2[orient_axis] = z_max
+
+        taper = SvLine.from_two_points(p1, p2).to_nurbs()
+        taper = taper.elevate_degree(target = max(profile_degree, 2))
+        return taper
+
+    def _get_orient_axis_idx(self):
+        return 'XYZ'.index(self.orient_axis)
+
+    def process(self):
+        if not any(socket.is_linked for socket in self.outputs):
+            return
```

Comment:
I don't think we need this check now. It makes update system more complex. It can be even less efficient because some nodes update even when there are no output links but update system have to update them anyway. Update system does not know whether the node has this check or no.
https://github.com/nortikin/sverchok/blob/3d76f45c9c2c6e423335c42f6ebc33392e9aad2b/core/main_tree_handler.py#L391-L396

Reply:
I'm afraid I don't understand.
If the node does not have connected outputs, why spend time doing complex calculations?

Reply:
Some nodes does not have such optimization. How update system should understand whether a node has this check or not to update the node properly?

Reply:
As far as I understand, the only case when the node has to be updated without having connected outputs - is the case when it has "side effects", such as drawing in the viewport or generating objects in scene. We have only few of such node classes.
So, such nodes could have special class property (`has_side_effects = True`), so that update system could recognize them and trigger their update even when they do not have connected outputs.
For other nodes, update should not be triggered when they do not have connected outputs.
If we will have such logic in update system, we can remove the check from the nodes.

Reply:
But for now, do I correctly understand that the update system triggers `process()` of the node even if does not have connected outputs?
In this case nodes which do complex calculations have to do the check themselves.

Reply:
Yes the `process` is called even there are no output connection of a node. I guess there are case when such update are expected for example SNL node could do something without passing any result to next nodes.
The thing is that such checks do not let make optimization in update system and adding this checks in new nodes make future improvements in this direction harder.

Reply:
But without this check node trees which include complex nodes do excessive computations. And at least for some nodes it means a lot of time.